Setup.hs

```diff
@@ -1,2 +1,6 @@
-import Distribution.Simple
-main = defaultMain
+module Main where
+
+import Distribution.Extra.Doctest (defaultMainWithDoctests)
```

Comment:
A note on this patch + doctests: because our doctests import `Control.Effect.Pure`, which imports `Control.Monad.Base`, we have to make sure that transformers-base is accessible to the GHC invocation that `doctests` performs. This is, unfortunately, buggy on versions of GHC older than 8.6. The only way I could find for it to work was to incur a dependency on `cabal-doctest`, which takes care of pulling in the correct `package-db` fields that we have to pass to the `doctest` invocation.

It is a shame that the `doctest` package is so fragile.

Reply:
Could you do me a quick favour and check whether you have multiple (ignored) `.ghc.environment…` files in the project dir? I’ve seen `doctest` fail for older versions of GHC when such were present, but succeed when there’s only one (for the matching version of GHC).

Reply:
Whoa, that could be it. Hang on, lemme check.

Reply:
Nope, that wasn’t it 😞